Lib/test/support/__init__.py

```diff
@@ -1991,5 +1991,7 @@ def check_disallow_instantiation(testcase, tp, *args, **kwds):
 
     See bpo-43916.
     """
-    msg = f"cannot create '{tp.__module__}\.{tp.__name__}' instances"
+    mod = tp.__module__
+    name = tp.__name__
+    msg = fr"cannot create '{mod if mod != 'builtins' else ''}.?{name}' instances"
```

Comment:
`'{mod if mod != 'builtins' else ''}.?` is complex and I dislike the optional dot in the regex. I understand that it's either mod+'.' or an empty string.

I suggest to add a qualname variable:

```
if mod != 'builtins':
  qualname = f'{mod}.{name}'
else:
  qualname = name
msg = fr"cannot create '{re.escape(qualname)}' instances"
```

I recommend `re.escape()` to avoid interpreting the dot as "any character".

Reply:
Yes, that's definitely nicer.